Source/Scene/ImplicitTileset.js

```diff
@@ -139,3 +191,49 @@ export default function ImplicitTileset(baseResource, tileJson) {
    */
   this.maximumLevel = extension.maximumLevel;
 }
+
+/**
+ * Gather JSON headers for all contents in the tile.
+ * This handles both regular tiles and tiles with the
+ * `3DTILES_multiple_contents` extension
+ *
+ * @param {Object} tileJson The JSON header of the tile with the 3DTILES_implicit_tiling extension.
+ * @return {Object[]} An array of JSON headers for the contents of each tile
+ * @private
+ */
+function gatherContentHeaders(tileJson) {
+  if (has3DTilesExtension(tileJson, "3DTILES_multiple_contents")) {
+    return tileJson.extensions["3DTILES_multiple_contents"].content;
+  }
+
+  if (defined(tileJson.content)) {
+    return [tileJson.content];
+  }
+
+  return [];
+}
+
+function makeTileHeaderTemplate(tileJson) {
+  var template = clone(tileJson, true);
+
+  // remove the implicit tiling extension to prevent infinite loops
+  delete template.extensions["3DTILES_implicit_tiling"];
+
+  // content is handled separately, so remove content-related properties
+  delete template.content;
+  delete template.extensions["3DTILES_multiple_contents"];
+
+  // if there are no other extensions, remove the extensions property to
+  // keep each tile simple
+  var removeExtensions = true;
+  for (var key in template.extensions) {
+    if (template.extensions.hasOwnProperty(key)) {
+      removeExtensions = false;
+    }
+  }
+  if (removeExtensions) {
+    delete template.extensions;
+  }
```

Comment:
More concise at the cost of an allocation (though I don't think a huge concern)

```suggestion
  if (Object.keys(template.extensions).length === 0) {
    delete template.extensions;
  }
```